### DISCUSSION TITLE
Play with DevOps - Jenkins, Docker, Nodejs, React and Postgres

### DISCUSSION THREAD

**Discussion by fera17-zz** [BretFisher/ama#138] [Q&A]
Hi Bret, just finished the https://www.udemy.com/docker-mastery/ course, brilliant! I have a question not directly linked to the course itself. I want now to play a little bit with docker, swarm, etc. And I am using a local linux machine (with a swarm of 3 nodes installed) for Dev and Test. And I will use AWS EC2 instances for Stage / Prod envs. (3 instances of the free tier in which I've already set up a swarm with 3 nodes). My objective is: trying to build a React / Nodejs / PostGres simple app to play with. I want to use as IDE Visual studio 2017 or Visual studio Code (they run in a Windows 10 laptop). And Jenkins / Github to orchestrate a CI/CD pipeline. Given that, what are your suggestions? Better VS 2017 or VSC? Do you know a way to connect the debugging of Visual studio (both versions) to the Docker Dev env? Any tips or suggestion related to everything I wrote it would be really appreciated! thanks, Fabio

**Comment by BretFisher** [comment]
I haven't used full VS IDE in some time. VS Code is great but as far as I know Full VS would have more features...? especially around IntelliSense and debugging, but not sure the feature difference. I think VS Code is awesome and use it.

Yes if you're talking about Node.js, My [Docker Node.js sample project](https://github.com/BretFisher/node-docker-good-defaults) has a VS Code config for debugging. If you're using docker-compose like that project shows, the debugging should "just work" in VS Code.
